src/emmeans.jl

```diff
@@ -0,0 +1,187 @@
+                 invlink=identity, levels=Dict(), dof=nothing)
+    form = formula(model)
+    typical = mean
+    # cat_terms = filter(x -> x isa CategoricalTerm, terms(form.rhs))
+    # defaults =  Dict(tt.sym => tt.contrasts.levels for tt in cat_terms)
+    defaults = Dict{Symbol,Vector}()
+    for tt in terms(get_matrix_term(form.rhs))
+        # this kinda feels like a place for dispatch but....
+        if tt isa CategoricalTerm
+            defaults[tt.sym] = tt.contrasts.levels
+        elseif tt isa ContinuousTerm
+            defaults[tt.sym] = [tt.mean]
+            # handle StandardizdPredictors and other wrapped Terms
+        elseif hasproperty(tt, :term) && tt.term isa ContinuousTerm
+            defaults[tt.term.sym] = [hasproperty(tt, :center) ? tt.center : tt.term.mean]
+        end
+    end
+    levels = merge(defaults, levels) # prefer user specified levels
+    grid = expand_grid(levels)
+    eff_col = string(something(eff_col, _responsename(model)))
+    err_col = string(err_col)
+
+    result = effects!(grid, model; eff_col, err_col, typical, invlink)
+    if !isnothing(dof)
+        result[!, :dof] .= _dof(dof, model)
+    end
+    return result
+end
+
+"""
+    empairs(model::RegressionModel; eff_col=nothing, err_col=:err,
+            invlink=identity, levels=Dict(), dof=nothing, padjust=identity)
+    empairs(df::AbstractDataFrame; eff_col, err_col=:err, padjust=identity)
+
+Compute pairwise differences of estimated marginal means.
+
+The method for `AbstractDataFrame` acts on the results of [`emmeans`](@ref),
+while the method for `RegressionModel` is a convenience wrapper that calls
+`emmeans` internally.
+
+Keyword arguments are generally same as `emmeans`.
+
+By default, pairs are computed for all combinations of categorical variables and
+the means/centers of continuous variables. The contrast for a pair "a" vs "b"
+is represented in the column for the contrasted variable with `a > b`. For
+variables that are the same within a pair (e.g., a continuous variable), the sole
+value is displayed as is.
+
+If `dof` is not `nothing`, then p-values are also computed using a t-distribution
+with the resulting degrees of freedom. The results for `dof=Inf` correspond to
+using a z-distribution, but the column names in the returned dataframe remain
+`t` and `Pr(>|t|)`.
+
+If `padjust` is provided, then it is used to compute adjust the p-values for
+multiple comparisons. [`MultipleTesting.jl`](https://juliangehring.github.io/MultipleTesting.jl/stable/)
+provides a number of useful possibilities for this.
+
+!!! warning
+    This feature is experimental and the precise column names and presentation of
+    contrasts/differences may change without being considered breaking.
+
+!!! warning
+    The use of `invlink` is subject to a number of interpretation subtleties.
+    The EM means are computed on the scale of the linear predictor, then
+    backtransformed to the transformed scale. The associated errors on the
```

Comment:
```suggestion
    The EM means are computed on the scale of the linear predictor then
    transformed to the scale of the response via the inverse link function.
    The associated errors on the
```

Reply:
This isn't quite right. If you had e.g. `log(y) ~ x` and passed `invlink=exp`, then you're back transforming to the "raw" scale, but technically the model response is `log(y)` and not `y` and so the response is on the scale of the linear predictor.

Reply:
Ah right, forgot about transformed responses